engine/runtime-language-arrow/src/test/java/org/enso/interpreter/arrow/VerifyArrowTest.java

```diff
@@ -0,0 +1,296 @@
+    assertEquals((byte) 47, v.asByte());
+    int8Array.setArrayElement(5, 21);
+    v = int8Array.getArrayElement(5);
+    assertEquals((byte) 21, v.asByte());
+    try {
+      int8Array.setArrayElement(5, 300);
+      fail("expected out of bounds exception");
+    } catch (UnsupportedOperationException e) {
+    }
+  }
+
+  @Test
+  public void castInt() {
+    var typeLength = LogicalLayout.Int32;
+    var testValues = new Object[] {3, 1, 5, 3};
+    try (BufferAllocator allocator = new RootAllocator();
+        BaseFixedWidthVector intVector =
+            allocateFixedLengthVector(allocator, testValues, typeLength); ) {
+      var int32Constr = ctx.eval("arrow", "cast[" + typeLength + "]");
+      assertNotNull(int32Constr);
+      Value int32Array =
+          int32Constr.newInstance(
+              intVector.getDataBufferAddress(),
+              intVector.getDataBuffer().capacity() / typeLength.sizeInBytes());
+
+      assertNotNull(int32Array);
+      for (int i = 0; i < testValues.length; i++) {
+        if (testValues[i] != null) {
+          assertEquals(testValues[i], int32Array.getArrayElement(i).asInt());
+        } else {
+          assertTrue(int32Array.getArrayElement(i).isNull());
+        }
+      }
+    }
+
+    testValues = new Object[] {3, null, 5, 3, 7, 18, null, 9, 7, null, null, 100};
+    try (BufferAllocator allocator = new RootAllocator();
+        BaseFixedWidthVector intVector =
+            allocateFixedLengthVector(allocator, testValues, typeLength); ) {
+      var int32Constr = ctx.eval("arrow", "cast[" + typeLength + "]");
+      assertNotNull(int32Constr);
+      Value int32Array =
+          int32Constr.newInstance(
+              intVector.getDataBufferAddress(),
+              intVector.getDataBuffer().capacity() / typeLength.sizeInBytes(),
+              intVector.getValidityBufferAddress());
+
+      assertNotNull(int32Array);
+      for (int i = 0; i < testValues.length; i++) {
+        if (testValues[i] != null) {
+          assertEquals(testValues[i], int32Array.getArrayElement(i).asInt());
+        } else {
+          assertTrue(int32Array.getArrayElement(i).isNull());
+        }
+      }
+
+      // Verify vector is memory-mapped, not copied
+      assertTrue(int32Array.getArrayElement(10).isNull());
+      ((IntVector) intVector).set(10, 12);
+      assertFalse(int32Array.getArrayElement(10).isNull());
+      assertEquals(12, int32Array.getArrayElement(10).asInt());
+    }
+
+    typeLength = LogicalLayout.Int64;
+    testValues = new Object[] {(long) 3, null, (long) 5, (long) 3};
+    try (BufferAllocator allocator = new RootAllocator();
+        BaseFixedWidthVector vector =
+            allocateFixedLengthVector(allocator, testValues, typeLength); ) {
+      var int32Constr = ctx.eval("arrow", "cast[" + typeLength + "]");
+      assertNotNull(int32Constr);
+      Value int32Array =
+          int32Constr.newInstance(
+              vector.getDataBufferAddress(),
```

Comment:
address and capacity is OK, for now. It'd be handy (for interaction with Python) to also support [InteropLibrary.isPointer](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/InteropLibrary.html#isPointer-java.lang.Object-).